rotkehlchen/chain/evm/decoding/cowswap/decoder.py

```diff
@@ -228,21 +251,28 @@ def _detect_relevant_trades(
                 )
                 decoded_events.append(spend_event)
 
-            fee_event = None
-            if swap_data.fee_amount != ZERO:
-                fee_event = self.base.make_event_next_index(
-                    tx_hash=transaction.tx_hash,
-                    timestamp=transaction.timestamp,
-                    event_type=HistoryEventType.TRADE,
-                    event_subtype=HistoryEventSubType.FEE,
-                    asset=swap_data.from_asset,
-                    balance=Balance(amount=swap_data.fee_amount),
-                    location_label=receive_event.location_label,
-                    notes=f'Spend {swap_data.fee_amount} {spend_event.asset.symbol_or_name()} as a cowswap fee',  # noqa: E501
-                    counterparty=CPT_COWSWAP,
-                    address=transaction.to_address,
+            if swap_data.fee_amount == ZERO:
+                raw_fee_amount, swap_data.order_type = self.cowswap_api.query_order_data(swap_data.order_uid)  # noqa: E501
```

Comment:
You are not handling the potential RemoteError here. This means that the entire decoding process will break down the moment the API does not respond for any reason. That's bad.You just have to catch it here and if there is an error just don't add a fee and not change order type. But log the error and state the problem in the logs.